### PR TITLE
Pick hosts for nodes to reduce resource allocation skew

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

**Review by freva — APPROVED** [review]
:1st_place_medal:

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/PrioritizableNode.java [hunk 1]
```diff
@@ -140,4 +186,19 @@ PrioritizableNode build() {
         }
     }
 
+    /** The mean and mean deviation (squared difference) of a bunch of numbers */
+    private static class Mean {
+
+        private double mean;
```
[the remark this answers is not in the document]
Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/PrioritizableNode.java]: Done, thanks!

**Review by kkraune — APPROVED** [review]
Great!

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/PrioritizableNodeTest.java [hunk 2]
```diff
@@ -18,32 +18,132 @@
...
+                node("2", node(1, 5), host(21, 20), host(40, 40)),
+                node("3", node(1, 5), host(20, 20), host(40, 40)),
+                node("4", node(1, 5), host(20, 22), host(40, 40))
+        );
+        assertOrder(expected);
+    }
+
+    private void assertOrder(List<PrioritizableNode> expected) {
         List<PrioritizableNode> copy = new ArrayList<>(expected);
         Collections.shuffle(copy);
```
Comment (mpolden) [inline node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/PrioritizableNodeTest.java]: Remove? Shuffle and then sort...

**Review by mpolden — COMMENTED** [review]
👍

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/PrioritizableNodeTest.java (hunk 2 above)
Comment (bratseth) [reply node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/PrioritizableNodeTest.java]: It's to test the order resulting from sort. Before shuffle it is already sorted correctly.

**Pull request merged**